src/main/java/org/opensearch/ad/indices/AnomalyDetectionIndices.java

```diff
@@ -161,26 +159,26 @@ class IndexState {
     /**
      * Constructor function
      *
-     * @param client         ES client supports administrative actions
-     * @param clusterService ES cluster service
+     * @param restClient         ES client supports administrative actions
+     * @param sdkClusterService ES cluster service
      * @param threadPool     ES thread pool
      * @param settings       ES cluster setting
      * @param nodeFilter     Used to filter eligible nodes to host AD indices
      * @param maxUpdateRunningTimes max number of retries to update index mapping and setting
      */
     public AnomalyDetectionIndices(
-        Client client,
-        ClusterService clusterService,
+        SDKRestClient restClient,
+        SDKClusterService sdkClusterService,
         ThreadPool threadPool,
         Settings settings,
         DiscoveryNodeFilterer nodeFilter,
         int maxUpdateRunningTimes
     ) {
-        this.client = client;
-        this.adminClient = client.admin();
-        this.clusterService = clusterService;
+        this.client = restClient;
+        this.adminClient = restClient;
+        this.clusterService = sdkClusterService;
         this.threadPool = threadPool;
-        this.clusterService.addLocalNodeMasterListener(this);
+        // this.clusterService.addLocalNodeMasterListener(this);
```

Comment:
We should create an issue to create a transport method for this listener as this class implements `LocalNodeMasterListener`.

Reply:
I'm under the impression this isn't needed at all.  But I'm happy to create an issue to determine whether to make it a no-op.